test/lib/Dialect/VectorExt/TestVectorMaskingUtils.cpp

```diff
@@ -35,11 +35,47 @@ struct TestVectorMaskingUtils
   }
 
   void runOnFunction() override {
-    getFunction().walk([](TiledLoopOp loopOp) {
-      OpBuilder builder(loopOp);
-      if (failed(predicateTiledLoop(builder, loopOp)))
-        loopOp.emitError("Predication of tiled loop failed");
-    });
+    // Try different testing approaches until one triggers the predication
+    // transformation for that particular function.
+    bool predicationSucceeded = false;
```

Comment:
Did you consider pass flags to control tiled loop vs function predication?

Reply:
Yes, the problem is that I have to add support for more ops and things will get complicated if we keep the tests in the same file (multiple RUN rules + different CHECK prefixes, etc.). We can revisit this when we add more tests or hit any limitation. Thanks for the feedback!

Reply:
Sounds good!